msbuild/Xamarin.iOS.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs

```diff
@@ -236,6 +246,73 @@ bool Compile (PDictionary plist)
 			return !Log.HasLoggedErrors;
 		}
 
+		void SetRequiredArchitectures (PDictionary plist)
+		{
+			PObject capabilities;
+
+			if (plist.TryGetValue (ManifestKeys.UIRequiredDeviceCapabilities, out capabilities)) {
+				if (capabilities is PArray) {
+					var architectureValues = new HashSet<string> (new[] { "armv6", "armv7", "arm64" });
+					var array = (PArray) capabilities;
+
+					// Remove any architecture values
+					for (int i = 0; i < array.Count; i++) {
+						var value = array[i] as PString;
+
+						if (value == null || !architectureValues.Contains (value.Value))
+							continue;
+
+						array.RemoveAt (i);
+					}
+
+					// If-and-only-if the TargetArchitecture is a single architecture, set it as a required device capability
+					switch (architectures) {
+					case TargetArchitecture.ARM64:
+						array.Add (new PString ("arm64"));
+						break;
+					case TargetArchitecture.ARMv7:
+						array.Add (new PString ("armv7"));
+						break;
+					}
+				} else if (capabilities is PDictionary) {
+					var dict = (PDictionary) capabilities;
+
+					switch (architectures) {
+					case TargetArchitecture.ARM64:
+						dict["arm64"] = new PBoolean (true);
+						dict.Remove ("armv6");
+						dict.Remove ("armv7");
+						break;
+					case TargetArchitecture.ARMv7:
+						dict["armv7"] = new PBoolean (true);
+						dict.Remove ("armv6");
+						dict.Remove ("arm64");
+						break;
+					default:
+						dict.Remove ("armv6");
+						dict.Remove ("armv7");
+						dict.Remove ("arm64");
+						break;
+					}
+				}
+			} else {
+				var array = new PArray ();
+
+				// If-and-only-if the TargetArchitecture is a single architecture, set it as a required device capability
+				switch (architectures) {
+				case TargetArchitecture.ARM64:
+					array.Add (new PString ("arm64"));
```

Comment:
Mmh this is a copy/paste of L270 above, let's make it a method (:

Reply:
It's similar but not the same.

Reply:
@jstedfast https://github.com/xamarin/xamarin-macios/blob/e6f4f4b72a6bd5c6a1c48e76a8faef70b683557e/msbuild/Xamarin.iOS.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs#L249-L256 == https://github.com/xamarin/xamarin-macios/blob/e6f4f4b72a6bd5c6a1c48e76a8faef70b683557e/msbuild/Xamarin.iOS.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs#L282-L290 to me (: